apps/web/src/utils/page-visibility.ts

```diff
@@ -17,37 +17,42 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-var hidden, visibilityChange;
-if (typeof document.hidden !== "undefined") {
-  // Opera 12.10 and Firefox 18 and later support
-  hidden = "hidden";
-  visibilityChange = "visibilitychange";
-} else if (typeof document.msHidden !== "undefined") {
-  hidden = "msHidden";
-  visibilityChange = "msvisibilitychange";
-} else if (typeof document.webkitHidden !== "undefined") {
-  hidden = "webkitHidden";
-  visibilityChange = "webkitvisibilitychange";
-}
-
-export function onPageVisibilityChanged(handler) {
+const getVisibilityChangeParams = () => {
+  if ("hidden" in document) {
+    // Opera 12.10 and Firefox 18 and later support
+    return ["hidden", "visibilityChange"] as const;
+  } else if ("msHidden" in document) {
+    return ["msHidden", "msvisibilityChange"] as const;
+  } else if ("webkitHidden" in document) {
+    return ["webkitHidden", "webkitvisibilityChange"] as const;
+  }
+  return ["hidden", "visibilityChange"] as const;
+};
+
+export function onPageVisibilityChanged(
+  handler: (
+    status: "online" | "offline" | "visibilitychange",
+    bool: boolean
+  ) => void
+) {
   onDeviceOnline(() => handler("online", false));
   onDeviceOffline(() => handler("offline", false));
 
   // Handle page visibility change
+  const { hidden, visibilityChange } = getVisibilityChangeParams();
```

Comment:
You are returning an array with 2 elements. It's not an object. So use `[hidden, ...]` instead.